File: backend/app/schemas/live.py
"""
直播Schema
"""
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field
from app.models.live_room import LiveStatus, MessageType


class LiveRoomBase(BaseModel):
    """直播间基础Schema"""
    title: str = Field(..., min_length=1, max_length=200, description="直播标题")
    description: Optional[str] = Field(None, description="直播描述")
    cover_image: Optional[str] = Field(None, description="封面图片")
    start_time: Optional[datetime] = Field(None, description="开始时间")


class LiveRoomCreate(LiveRoomBase):
    """创建直播间Schema"""
    pass


class LiveRoomUpdate(BaseModel):
    """更新直播间Schema"""
    title: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    cover_image: Optional[str] = None
    status: Optional[LiveStatus] = None
    start_time: Optional[datetime] = None
    end_time: Optional[datetime] = None


class LiveRoomResponse(LiveRoomBase):
    """直播间响应Schema"""
    id: int
    teacher_id: int
    push_url: Optional[str]
    pull_url_rtmp: Optional[str]
    pull_url_flv: Optional[str]
    pull_url_hls: Optional[str]
    status: LiveStatus
    end_time: Optional[datetime]
    viewer_count: int
    max_viewer_count: int
    created_at: datetime
    updated_at: datetime
    
    # 关联数据
    teacher_name: Optional[str] = None
    
    class Config:
        from_attributes = True


class LiveChatMessageCreate(BaseModel):
    """创建聊天消息Schema"""
    live_room_id: int = Field(..., description="直播间ID")
    message: str = Field(..., min_length=1, description="消息内容")


class LiveChatMessageResponse(BaseModel):
    """聊天消息响应Schema"""
    id: int
    live_room_id: int
    user_id: int
    message: str
    message_type: MessageType
    created_at: datetime
    
    # 关联数据
    username: Optional[str] = None
    user_avatar: Optional[str] = None
    
    class Config:
        from_attributes = True

File: frontend/src/services/live.ts
import api from './api'

export interface LiveRoom {
  id: number
  title: string
  description: string
  teacher_id: number
  cover_image: string
  status: 'not_started' | 'living' | 'finished'
  start_time: string
  end_time?: string
  viewer_count: number
  max_viewer_count: number
  push_url?: string
  pull_url_rtmp?: string
  pull_url_flv?: string
  pull_url_hls?: string
}

export interface LiveListParams {
  page?: number
  page_size?: number
  status?: string
}

export const liveApi = {
  // 获取直播列表
  getLives(params?: LiveListParams) {
    return api.get('/api/v1/lives', { params })
  },

  // 获取直播详情
  getLiveDetail(id: number) {
    return api.get(`/api/v1/lives/${id}`)
  },

  // 创建直播
  createLive(data: Partial<LiveRoom>) {
    return api.post('/api/v1/lives', data)
  },

  // 更新直播
  updateLive(id: number, data: Partial<LiveRoom>) {
    return api.put(`/api/v1/lives/${id}`, data)
  },

  // 删除直播
  deleteLive(id: number) {
    return api.delete(`/api/v1/lives/${id}`)
  },
}
